src-cpp/app/modules/main/wallet/accounts/view.cpp

```diff
@@ -20,8 +23,120 @@ Model* View::getModel()
     return m_modelPtr;
 }
 
-void View::setModelItems(QVector<Item> &accounts) {
+void View::setModelItems(QVector<Item*> accounts) {
     m_modelPtr->setItems(accounts);
+    m_currentAccountPtr->setData(accounts.at(0));
     modelChanged();
 }
+
+void View::refreshWalletAccounts()
+{
+    auto walletAccounts = m_controllerPtr->getWalletAccounts();
+
+    if(walletAccounts.size() > 0)
+    {
+        QVector<Item*> items;
+        foreach(const auto& acc, walletAccounts)
+        {
+            items << new Item(acc.name, acc.address, acc.path, acc.color, acc.publicKey, acc.walletType, acc.isWallet, acc.isChat, 0, this);
```

Comment:
you can create a factory method like `Item::createFromAccount(walletAccount)`

Reply:
Would have to expose the DTO type to item in this case? is it really needed?